packages/lib/utils/recall.ts

```diff
@@ -0,0 +1,100 @@
+import { TSurvey, TSurveyQuestion } from "@formbricks/types/surveys";
+
+export function extractId(text: string) {
+  const pattern = /recall:([A-Za-z0-9]+)/;
+  const match = text.match(pattern);
+  if (match && match[1]) {
+    return match[1];
+  } else {
+    return null;
+  }
+}
+
+export function extractIds(text: string) {
+  const pattern = /recall:([A-Za-z0-9]+)/g;
+  const matches = text.match(pattern);
+  if (!matches) {
+    return [];
+  }
+  return matches
+    .map((match) => {
+      const matchPattern = /recall:([A-Za-z0-9]+)/;
+      const idMatch = match.match(matchPattern);
+      return idMatch ? idMatch[1] : null;
+    })
+    .filter((id) => id !== null);
+}
+
+export function extractFallbackValue(text: string): string {
+  const pattern = /fallback:(\S*)/;
+  const match = text.match(pattern);
+  if (match && match[1]) {
+    return match[1];
+  } else {
+    return "";
+  }
+}
+
+export function extractRecallInfo(headline: string): string | null {
+  const pattern = /recall:([A-Za-z0-9]+)\/fallback:(\S*)/;
+  const match = headline.match(pattern);
+  return match ? match[0] : null;
+}
+
+export function findRecallInfoById(text: string, id: string): string | null {
```

Comment:
GPT note:
GPT note:

Ensure that your regular expressions correctly handle special characters. For example, if id in `findRecallInfoById` can contain special characters that have meaning in regular expressions (like . or *), they should be escaped properly.

Reply:
Tested it with special symbols and it works fine 😊